### PR TITLE
[PBF-452/PBF-453/PBF475] Bitrate implementation for OSMF plugin

### PR DESCRIPTION
**Pull request by nagavenivn**

### CODE REVIEW

File: src/osmf/js/osmf_flash.js [hunk 1]
```diff
@@ -573,6 +575,20 @@ require("../../../html5-common/js/utils/constants.js");
                              { "isFullScreen" : false, "paused" : event.target.paused });
     };
 
+    var raiseBitratesAvailable = function(event) {
+     var vtcBitrates = [{id: "auto", width: 0, height: 0, bitrate: 0 }];
+        for (var i in event.eventObject) {
+          var vtcBitrate = {
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: indentations are a little off here - not a real issue, just noticed

File: src/osmf/lib/HDSPlayer.as [hunk 2]
```diff
@@ -851,6 +852,42 @@ package
...
     /**
+     * Provides the total available bitrates and dispatches BITRATES_AVAILABLE event.
+     * @public
+     * @method HDSPlayer#totalBitratesAvailable
+     */
+    public function  totalBitratesAvailable():void
+    { 
+      var eventObject:Object=new Object();
+      
+      if (getStreamsCount() > 0)
```
Comment (angieisrani) [inline src/osmf/lib/HDSPlayer.as]: i think this if is not needed

File: src/osmf/js/osmf_flash.js [hunk 3]
```diff
@@ -94,7 +94,8 @@ require("../../../html5-common/js/utils/constants.js");
     }
     this.encodings = testForFlash();
     this.technology = OO.VIDEO.TECHNOLOGY.FLASH;
-    this.features = [ OO.VIDEO.FEATURE.CLOSED_CAPTIONS ];
+    this.features = [ OO.VIDEO.FEATURE.CLOSED_CAPTIONS,
+                      OO.VIDEO.FEATURE.BITRATE_CONTROL ];
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: the bitrate control feature is not supported unless the plugin allows the player to set a bitrate and i didn't see the logic for that below

**Comment by angieisrani** [comment]
Do we  need to raise the list of available bitrates if we don't support setting them?

File: src/osmf/js/osmf_flash.js [hunk 4]
```diff
@@ -573,6 +575,20 @@ require("../../../html5-common/js/utils/constants.js");
...
+        for (var i in event.eventObject) {
+          var vtcBitrate = {
+          id: event.eventObject[i].id,
+          width: event.eventObject[i].width,
+          height: event.eventObject[i].height,
+          bitrate: event.eventObject[i].bitrate
+        }
+        vtcBitrates.push(vtcBitrate);
+      }
+      newController.notify(newController.EVENTS.BITRATES_AVAILABLE,vtcBitrates);
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: this does not save the bitrates anywhere for usage by id later

**Comment by nagavenivn** [comment]
@angieisrani:  Pushed all changes related to bitrate to this branch as discussed with you.

File: src/osmf/js/osmf_flash.js [hunk 5]
```diff
@@ -310,6 +313,19 @@ require("../../../html5-common/js/utils/constants.js");
     };
 
     /**
+     * Sets the stream to play back based on given stream ID. Plugin must support the
+     * BITRATE_CONTROL feature to have this method called.
+     * @public
+     * @method TemplateVideoWrapper#setBitrate
```
Comment (angieisrani) [inline src/osmf/js/osmf_flash.js]: small issue - needs correct class name

**Comment by angieisrani** [comment]
LGTM

**Pull request merged**